src/main/osd/osd_elements.c

```diff
@@ -601,6 +604,54 @@ static void osdElementArtificialHorizon(osdElementParms_t *element)
 
     element->drawElement = false;  // element already drawn
 }
+
+static void osdElementUpDownReference(osdElementParms_t *element){
+// Up/Down reference feature displays reference points on the OSD at Zenith and Nadir
+// The positions are absolute and immutable by the user for consistent operation
+    if(abs(attitude.values.pitch)>450){ //pitch is greater than +45 or less than -45
+        float earthUpinBodyFrame[3] = {-rMat[2][0], -rMat[2][1], -rMat[2][2]}; //transforum the up vector to the body frame
+
+        float thetaB; // pitch from body frame to up vector
+        int psiBSign; // psi from body frame to up vector
+        char buf[2]; // character buffer
+
+        if(attitude.values.pitch>0.0){ //nose down
+            thetaB = -asinf(earthUpinBodyFrame[2]) * (180.0f / M_PIf); // get theta
+            psiBSign = -1; // correct the sign
```

Comment:
compute `psiB` here (BTW: using `float` for  `psiBSign` will probably result in faster code)